hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/cluster/SparkExecuteClusteringCommitActionExecutor.java

```diff
@@ -247,8 +247,8 @@ protected String getCommitActionType() {
    */
   private HoodieRecord<? extends HoodieRecordPayload> transform(IndexedRecord indexedRecord) {
     GenericRecord record = (GenericRecord) indexedRecord;
-    String key = record.get(HoodieRecord.RECORD_KEY_METADATA_FIELD).toString();
-    String partition = record.get(HoodieRecord.PARTITION_PATH_METADATA_FIELD).toString();
+    String key = populateMetaFields ? record.get(HoodieRecord.RECORD_KEY_METADATA_FIELD).toString() : keyGeneratorOpt.get().getRecordKey(record);
```

Comment:
this ternary switch is kind of everywhere. single method call like `fetchKey(populateMetaFields, record, keyGeneratorOpt)`  would be nice